### PR TITLE
Make Rados ready for PyPI

### PR DESCRIPTION
**Pull request by onyb**
Reopening the former pull request https://github.com/ceph/ceph/pull/9332 again, since it got messed up pretty badly trying to remove a file from the commit history.

This is also same as the `wip-rados-pypi` of `ceph.git`, but with reworded commit messages and `rados.c` nuked out from the history.

### CODE REVIEW

**Comment by jdurgin** [comment]
this is looking good to me, @tchaikov did you want to look it over too?

File: src/pybind/rados/setup.py [hunk 1]
```diff
@@ -61,6 +69,52 @@ def get_python_flags():
     }
 
 
+def check_sanity():
+    """
+    Test if development headers and library for rados is available by compiling a dummy C program.
+    """
+
+    tmp_dir = tempfile.mkdtemp(dir=os.path.dirname(__file__))
```
Comment (tchaikov) [inline src/pybind/rados/setup.py]: nit, would be better if we can respect `$TMPDIR`.

File: src/pybind/rados/setup.py [hunk 2]
```diff
@@ -122,6 +122,22 @@ def check_sanity():
     def cythonize(x, **kwargs):
         return x
 
+if '--without-cython' in sys.argv:
```
Comment (tchaikov) [inline src/pybind/rados/setup.py]: under what condition, user would use `setup.py` without cython? and we don't have `rados.c` in our repo yet, right?

File: src/pybind/rados/setup.py [hunk 3]
```diff
@@ -90,12 +88,20 @@ def check_sanity():
     compiler = new_compiler()
     customize_compiler(compiler)
 
+    if set(['MAKEFLAGS', 'MFLAGS', 'MAKELEVEL']).issubset(set(os.environ.keys())):
+        # The setup.py has been invoked by a top-level Ceph make.
+        # Set the appropriate CFLAGS and LDFLAGS
+        CEPH_ROOT = subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).strip().decode('utf-8')
+
+        compiler.add_include_dir(os.path.join(CEPH_ROOT, 'src', 'include'))
```
Comment (tchaikov) [inline src/pybind/rados/setup.py]: `CFLAGS` and `LDFLAGS` are passed to `setup.py` explicitly in `Distutils.cmake` and `Makefile.am`, i think they would ready the `compiler` to do its job? iirc, distutils do check the env variables to figure out the cflags and ldflags.

Comment (tchaikov) [inline src/pybind/rados/setup.py]: okay, probably `customize_compiler()` is different from unixcompiler...

**Comment by hansbogert** [comment]
Sorry for necromancing an old issue, but I think my question applies here. I am not sure if I misunderstand the intent of all the issues related to pypi, but wasn't the whole idea to have updated versions of 'python-cephlibs' in pypi?

**Pull request merged**